projects/cds/src/merchandising/cms-components/merchandising-carousel/merchandising-carousel.component.ts

```diff
@@ -54,7 +50,10 @@ export class MerchandisingCarouselComponent {
     map(merchandsingProducts =>
       this.mapMerchandisingProductsToCarouselModel(merchandsingProducts)
     ),
-    filter<MerchandisingCarouselModel>(Boolean)
+    filter<{
```

Comment:
There's a type safety in rxjs' filter method signature, which:
- forces you to use this very boilerplate-like syntax
- prevents you to use short-hand Boolean constructor

An alternative is to use it like this: `filter(data => Boolean(data))`

Reply:
But now that I'm thinking more about it, the `data` is always going to be truthy, right?
Maybe remove the `filter` in this case.

Looking at the mapping methods:
- `getCarouselMetadata()` - should return an empty in the worst case scenario when there's no data
- `mapMerchandisingProductsToCarouselModel()` - returns an empty observable in the worst case scenario

Therefore, I think we don't need to perform null-check for neither of properties, correct?

Reply:
:+1: Removed it, tests pass and carousel renders